skimage/draw/draw.py

```diff
@@ -163,7 +166,38 @@ def set_color(img, coords, color):
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 1]], dtype=uint8)
 
     """
-
     rr, cc = coords
-    rr, cc = _coords_inside_image(rr, cc, img.shape)
-    img[rr, cc] = color
+
+    if alpha is None:
+        rr, cc = _coords_inside_image(rr, cc, img.shape)
+    else:
+        rr, cc, alpha = _coords_inside_image(rr, cc, img.shape, val=alpha)
+
+    if not np.isscalar(color):
```

Comment:
Is this `if` necessary? If a scalar colour is interpreted as a gray level, this should be documented in the docstring.